updater/src/updater.rs

```diff
@@ -110,60 +158,80 @@ pub struct Updater {
 
 const CLIENT_ID: &'static str = "parity";
 
-fn client_id_hash() -> H256 {
-	CLIENT_ID.as_bytes().into()
+lazy_static! {
+	static ref CLIENT_ID_HASH: H256 = CLIENT_ID.as_bytes().into();
 }
 
-fn platform() -> String {
-	if cfg!(target_os = "macos") {
-		"x86_64-apple-darwin".into()
-	} else if cfg!(windows) {
-		"x86_64-pc-windows-msvc".into()
-	} else if cfg!(target_os = "linux") {
-		format!("{}-unknown-linux-gnu", Target::arch())
-	} else {
-		version::platform()
-	}
+lazy_static! {
+	static ref PLATFORM: String = {
+		if cfg!(target_os = "macos") {
+			"x86_64-apple-darwin".into()
+		} else if cfg!(windows) {
+			"x86_64-pc-windows-msvc".into()
+		} else if cfg!(target_os = "linux") {
```

Comment:
If this is going to be backported, we might encounter problems with different openssl versions on different distros. See three different version of linux packages [here](https://github.com/paritytech/parity/releases/tag/v1.9.4) for example.

How do we handle that in out current updater? We might've been breaking clients' installations on Debian and CentOS in the worst case.

Reply:
I don't think we handle it at all, AFAIK only one Linux release is published in the Operations contract for a given platform. Maybe we can start detecting the distro and treat those as different platforms: "x64-ubuntu-linux-gnu"/"x64-centos-linux-gnu" etc. I think this will require changes in https://github.com/paritytech/push-release and in our CI pipeline.